include/alpaka/core/DemangleTypeNames.hpp

```diff
@@ -1,23 +1,90 @@
+    template<typename T>
+    consteval std::string_view embedTypeName()
+    {
+#if defined(__GNUC__) || defined(__clang__) || defined(__PGI) || defined(__NVCOMPILER)
+        // gcc, clang, PGI, nvc++
+        return __PRETTY_FUNCTION__;
+#elif defined(_MSC_VER)
+        // MSVC
+        return __FUNCSIG__;
+#else
+        // unknown compiler, use the c++20 standard function_name()
+        return std::source_location::current().function_name();
+#endif
+    }
+
+    // Convert a string_view to a null-terminated array of bytes.
+    template<std::size_t... Idxs>
+    consteval auto substringAsArray(std::string_view str, std::index_sequence<Idxs...>)
+    {
+        // Add null termination
+        return std::array{str[Idxs]..., '\0'};
+    }
+
+    // Use a fundamental type to detect the prefix and suffix, because MSVC includes the class, struct, or union
+    // keyword in the type signature.
+    using TestPatternType = double;
 
-#include <boost/core/demangle.hpp>
+    constexpr std::string_view testPatternName("double");
+
+    // Embed the demangled type name into a function signature, then extract it and return it as a null-terminated
+    // array of bytes.
+    template<typename T>
+    consteval auto demangleAsArray()
+    {
+        constexpr std::string_view testPatternText = embedTypeName<TestPatternType>();
+        constexpr size_t prefixSize = testPatternText.find(testPatternName);
+        constexpr size_t suffixSize = testPatternText.size() - prefixSize - testPatternName.size();
+        constexpr std::string_view text = embedTypeName<T>();
+        constexpr size_t start = prefixSize;
+        constexpr size_t end = text.size() - suffixSize;
+        static_assert(start < end);
+        constexpr size_t length = end - start;
+        constexpr std::string_view name = text.substr(start, length);
+        return substringAsArray(name, std::make_index_sequence<name.size()>{});
+    }
+
+    // Store the demangled type name as a null-terminated array of bytes.
+    // Note: this could be a function-static constexpr variable in c++23.
+    template<typename T>
+    inline constexpr auto storage = demangleAsArray<T>();
+
+    // Return the demangled type name.
+    template<typename T>
+    consteval std::string_view demangle()
+    {
+        return std::string_view{storage<T>.data(), storage<T>.size()};
+    }
+
+} // namespace
 
 namespace alpaka::core
 {
-#if BOOST_COMP_CLANG
-#    pragma clang diagnostic push
-#    pragma clang diagnostic ignored "-Wexit-time-destructors"
-#    pragma clang diagnostic ignored "-Wmissing-variable-declarations"
-#endif
+
+    // constexpr template string_view that contains the demangled type name.
     template<typename T>
-    inline const std::string demangled = boost::core::demangle(typeid(T).name());
-#if BOOST_COMP_CLANG
-#    pragma clang diagnostic pop
-#endif
+    inline constexpr std::string demangled = std::string(demangle<T>());
```

Comment:
Note that using an `std::string_view` rather than an `std::string` _might_ be more efficient - but it requires changes in many other places, and can be postponed to a separate PR.